src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getTodos, getUser } from './api';
+
+enum Status {
+  ALL = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState(true);
+
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [loadingUser, setLoadingUser] = useState(true);
+
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.ALL);
+
+  const [query, setQuery] = useState('');
+
+  const handleSelectedStatus = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setSelectedStatus(event.target.value as Status);
+  };
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const showSelectedTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+
+    getUser(todo.userId)
+      .then(setSelectedUser)
+      .finally(() => setLoadingUser(false));
+  };
+
+  const closeSelectedTodo = () => {
+    setSelectedTodo(null);
+    setSelectedUser(null);
+  };
+
+  const resetStatus = () => {
+    setQuery('');
+    setSelectedStatus(Status.ALL);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoadingTodo(false));
+  });
```

Comment:
The `useEffect` hook is missing a dependency array. This will cause the effect to run after every render, which might not be the intended behavior. If you want it to run only once, pass an empty array as the second argument.